src/posts/introducing-pint-xarray/index.md

```diff
@@ -0,0 +1,292 @@
+These various packages work by providing a numerical array type that acts similarly to a NumPy array, and is intended to plug in and replace the raw NumPy array (a so-called "duck array type").
+
+Pint provides the `Quantity` object, which is a normal numpy array combined with a `pint.Unit`:
+
+```python
+q = np.array([6, 7]) * pint.Unit('metres')
+print(repr(q))
+```
+
+```
+Out: <Quantity([6 7], 'meter')>
+```
+
+Pint Quantities act like NumPy arrays, except that the units are carried around with the arrays, propagated through operations, and checked during operations involving multiple quantities.
+
+## Xarray now wraps Pint
+
+Thanks to the [tireless work](https://github.com/pydata/xarray/issues/3594) of xarray core developer Justus Magin, you can now enjoy this automatic unit-handling in xarray!
+
+Once you create a unit-aware Xarray object (see below for how) you can see the units of the data variables displayed as part of the printable representation.
+You also immediately get the key benefits of Pint:
+
+1. Units are propagated through arithmetic, and new quantities are built using the units of the inputs:
+
+   ```python
+   distance = xr.DataArray(10).pint.quantify("metres")
+   time = xr.DataArray(4).pint.quantify("seconds")
+
+   distance / time
+   ```
+
+   ```
+   Out:
+   <xarray.DataArray ()>
+   <Quantity(2.5, 'meter / second')>
+   ```
+
+2. Dimensionally inconsistent units are caught automatically:
+
+   ```python
+   apples = xr.DataArray(10).pint.quantify("kg")
+   oranges = xr.DataArray(200).pint.quantify("cm^3")
+
+   apples + oranges
+   ```
+
+   ```
+   Out:
+   DimensionalityError: Cannot convert from 'kilogram' ([mass]) to 'centimeter ** 3' ([length] ** 3)
+   ```
+
+3. Unit conversions become simple:
+
+   ```python
+   walk = xr.DataArray(500).pint.quantify('miles')
+
+   walk.pint.to('parsecs')
+   ```
+
+   ```
+   Out:
+   <xarray.DataArray ()>
+   <Quantity(2.6077643524162074e-11, 'parsec')>
+   ```
+
+4. You can specify that custom functions should expect certain units, and convert them if needed:
+
+   ```python
+   def jpl_trajectory_code(impulse):
+       """This function will only compute the correct result if supplied input in units of Newton-seconds."""
+
+       if impulse.pint.units != "newton * second":
+           impulse = impulse.pint.to("Newton * seconds")
+
+       # do some rocket science
+       ...
+
+   lockheed_impulse_value = xr.DataArray(5).pint.quantify("force_pounds * seconds")
+
+   jpl_trajectory_code(lockheed_impulse_value)
+   ```
+
+   ```
+   Out:
+   pint.DimensionalityError
+   ```
+
+   (Note: We are adding [new features](https://github.com/xarray-contrib/pint-xarray/pull/143) to make unit specification of function arguments more slick.)
```

Comment:
This should probably be something like
```suggestion
Note: We are adding [new features](https://github.com/xarray-contrib/pint-xarray/pull/143) to make specifying the units of  parameters of existing library functions for more slick.
```
now that we are agreeing on making `expects` a way to add unit support to function libraries that don't handle units (essentially, a "unsafe mode")

Reply:
I'm happy with the first suggestion.